sdl2-hyper-sonic-drivers/src/HyperSonicDrivers/drivers/midi/opl/OplDriver.cpp

```diff
@@ -17,36 +17,56 @@ namespace HyperSonicDrivers::drivers::midi::opl
 
     // TODO: allocateVoice and getFreeSlot should be merged into 1 function
 
-    OplDriver::OplDriver(const std::shared_ptr<hardware::opl::OPL>& opl,
-        const std::shared_ptr<audio::opl::banks::OP2Bank>& op2Bank,
-        const audio::mixer::eChannelGroup group,
-        const uint8_t volume,
-        const uint8_t pan) :
-        m_opl(opl), m_op2Bank(op2Bank), m_opl3_mode(opl->type == OplType::OPL3),
+    // TOOD m_opl3_mode rename to m_opl2_mode and invert is logic
```

Comment:
this todo

Reply:
can't due as oplwrite doesn't work well in dual opl2 mode. so it is strict an opl3 or nothing at the moment for stereo.